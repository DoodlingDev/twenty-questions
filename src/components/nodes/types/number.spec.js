/* eslint require-jsdoc: "off" */
import React from "react";
import FormNodeNumber from "../../../../src/components/nodes/types/number";
import { /* shallow, */ mount } from "enzyme";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

const defaultTestProps = {
  name: "test_number",
  path: "test.test_number",
  valueManager: {
    values: {
      "test.test_number": false,
    },
    validate: {},
    update: jest.fn(),
  },
  register: jest.fn(),
  label: "test number",
  description: "this is a test number",
};

function setup(renderFn, testProps = defaultTestProps) {
  return renderFn(<FormNodeNumber {...testProps}/>);
}

describe("rendering", () => {
  it("renders string types correctly", () => {
    const render = renderer.create(
      <FormNodeNumber
        name="test_number"
        path="test.test_number"
        valueManager={{
          values: {
            "test.test_boolean": false,
          },
          validate: {}
        }}
        register={() => {}}
        label="test number"
        description="this is a test number"
      />
    ).toJSON();
    expect(render).toMatchSnapshot();
  });
});

describe("change actions", () => {
  it("should properly call on update", () => {
    const wrapper = setup(mount);
    wrapper.find("input").simulate("change");
    expect(wrapper.props().valueManager.update.mock.calls.length).toBe(1);
  });
});
