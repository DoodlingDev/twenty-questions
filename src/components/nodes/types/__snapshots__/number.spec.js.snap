// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders string types correctly 1`] = `
<div
  className="q20_node"
>
  <div
    className="q20_node__testNumber--labelAndDescription"
  >
    <label
      className="q20_node__testNumber--label"
      htmlFor="test.test_number"
    >
      test number
      <span
        className="q20_node__testNumber--description"
        htmlFor="test.test_number"
      >
        this is a test number
      </span>
    </label>
  </div>
  <div
    className="q20_testNumber__errorHandler"
  >
    <input
      className="q20_nodeNumber__testNumber--input"
      id="test.test_number"
      onChange={[Function]}
      placeholder={undefined}
      type="number"
      value=""
    />
  </div>
</div>
`;
